### DISCUSSION TITLE
does keystatic have plans for editing multi-language content?

### DISCUSSION THREAD

**Discussion by stefanprobst** [Thinkmill/keystatic#460] [Q&A]
basically, i am looking for a way to use one content collection to manage both `posts/en/*` and `posts/de/*`.

netlify-cms allowed editing translated content side-by-side, and i was wondering if the keystatic team had similar/different ideas how to approach handling i18n content?

**Comment by JedWatson** [comment]
Hey @stefanprobst thanks for bringing this up. I don't think we have specific thoughts on it yet, but it's definitely something we should make possible. I'll mark this with the roadmap label so that we keep it on the radar, and if you (or anyone else reading this) have any other suggestions or requirements / prior art we should take into account, please do share!

**Comment by JedWatson** [comment]
Just to add an update here, this is still planned but the design isn't finalised at this stage, here's some details I was just discussing in the Astro discord:

> I've been wondering about how the UI should work actually... there might be a few places it should go.
> First thought is in the list screen (probably up near the Search field) so you can filter the list to all / one language... some way of making it clear which slugs exist in one / several / all languages seems useful.
> And then on the entry editing screen, probably some way of indicating which languages exist for the current slug (with an easy way to create missing languages)
> That sort of assumes that you're planning to have the same entries in multiple languages (translated), it maybe should work differently if you're planning to have completely different entries for different languages in the same content collection (that seems like a less common use case though?)

Aside from creating a specific UI for multi-language support, managing posts in multiple languages **does work today**.

We've added docs for using [nested slugs](https://keystatic.com/docs/path-configuration#nested-slug-example) which means in the Admin UI, you can provide both the subfolder and filename in the slug field.

So, if you want your content to be stored here:

```
content/blog/en/hello-world.mdoc
```

And your path configuration is this:

```
content/blog/**
```

In the UI, if you create a new post with the slug `en/hello-world` then it will work as expected.

**Comment by TajAlasfiyaa** [comment]
is there any update

**Comment by zanhk** [comment]
Is there an expected release date (even if broad, like Q3 2024, etc...) for this feature? 

Thanks

**Comment by simonswiss** [comment]
Collating this in https://github.com/Thinkmill/keystatic/issues/1080 — will try gather all mentions of i18n support, there are definitely a few :)
